src/snowflake/cli/api/sql_execution.py

```diff
@@ -42,9 +42,8 @@
 from snowflake.connector.errors import ProgrammingError
 
 
-class SqlExecutor:
-    def __init__(self, connection: SnowflakeConnection | None = None):
-        self._snowpark_session = None
+class BaseSqlExecutor:
+    def __init__(self, connection: SnowflakeConnection | None = None) -> None:
```

Comment:
IMO the connection should be a param of the concrete `SqlExecutor`, since other implementations might not need one (like a `TestSqlExecutor` for example)

Reply:
I think the point is this _is_ a concrete executor. Just a restricted one that doesn't overlap with the SQL facade in its contract.

Reply:
Just wanted to clarify the desired end state of this project

In my head based on previous conversations, we want the following abstractions:
1. One to execute all the raw SQL which we can mock or switch out the implementation for if the API for calling SQL changes, e.g. aforementioned REST APIs, and 
2. Another abstraction on top of that that only executes business logic, which doesn't need to change depending on how we execute queries and we can test/reason about easily

Am I correct in the classification that:
- BaseSqlExecutor = 1. (Data Access/Repository Layer)?
- SnowflakeSQLFacade = 2. (Business Logic/Service Layer)?

However I see that SnowflakeSQLFacade is still putting together raw SQL queries with strings, just passing them to the executor, and I'm a bit confused on the separation between these classes, since it seems now that SnowflakeSQLFacade and SQLExecutor are doing similar jobs in that they both extend BaseSQLExecutor's functionality, but with individual flavours of API definitions and error handling?

Reply:
the main difference between SqlExecutor and SnowflakeSQLFacade is in those API definitions and error handling. 
SnowflakeFacade defines the interactions from cli (or nativeapps for now) with snowflake on a higher level than `execute_query` and `use object`.
Currently we are managing sql calls (their logic, formulation, and their context e.g. role, wh,db) embedded within our logic whereas with SnowflakeFacade it's all encapsulated in its own layer. For an example, see how `execute_user_script` from SQLFacade is used VS how `execute_post_deploy_hooks` are written and used.

Reply:
Marcus, 1 & 2 are the same to me, they're both encapsulated in the facade. BaseSQLExecutor is very SQL-specific, and is there mostly so that we can add convenience methods to manipulate raw query results.